### PR TITLE
[Jest]  Migrate `Hash.test.js`

### PR DESCRIPTION
**Pull request by astridx**
I feel unsure about the migration and would like to coordinate something with you here.

1. Is the line `import {setPerformance, createMap as globalCreateMap, setWebGlContext} from '../util/test/util';` OK or should we change this because of the problems in https://github.com/maplibre/maplibre-gl-js/pull/583#issuecomment-961242487 ? 
In the latter case, I have to redo some other PRs. But I think with the migration of the attribution control, the problem is the `timer` variable, which is handled differently, right?

2. I changed the line `expect(map.getBearing()).toBe(0);`  to `expect(map.getBearing() === 0 ? 0 : map.getBearing()).toBe(0);` because of the signed zero problem. I did it in the same way as here: https://github.com/maplibre/maplibre-gl-js/pull/465  . Is this solution fine?
3. In some of the tests in this file I find the last line ` window.location.hash = '';` I did not change them because I wanted to change as little as possible. Besides, I'm sure it has a reason. But this reason can only be that something is set for future tests. This should not be in a test, right? Or am I missing something?

### CODE REVIEW

File: src/ui/hash.test.ts [hunk 1]
```diff
@@ -9,16 +13,16 @@ describe('hash', () => {
...
         Object.defineProperty(container, 'clientHeight', {value: 512});
-        return globalCreateMap(t, {container});
+        return globalCreateMap({container}, undefined);
     }
 
     test('#addTo', () => {
-        const map = createMap(t);
-        const hash = createHash();
+        const map = createMap();
+        const hash = createHash(undefined);
```
Comment (HarelM) [inline src/ui/hash.test.ts:25]: Make createHash parameter optional.

**Comment by HarelM** [comment]
Can you elaborate on 2? Seems like there should a "better looking" solution..

File: src/ui/hash.test.ts (hunk 1 above)
Comment (astridx) [reply src/ui/hash.test.ts]: Do you mean it like here: https://github.com/maplibre/maplibre-gl-js/pull/619/commits/6a19d3d25f58ca97f6a692b6a534d8de83aefd6c ?

**Comment by astridx** [comment]
> Can you elaborate on 2? Seems like there should a "better looking" solution..

My point is that the handling of the signed 0 is always done in the same way in this repo.  There are several workarounds. For example:
https://stackoverflow.com/questions/48405174/how-to-make-jest-not-distinguish-between-negative-zero-and-positive-zero

I would use my solution here. But if you prefer another solution, I'll be happy to change it.

File: src/ui/hash.test.ts (hunk 1 above)
Comment (HarelM) [reply src/ui/hash.test.ts]: Yup

**Comment by HarelM** [comment]
I thought this was something else. Negative zero is annoying...

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
Is this setup in draft mode?

**Comment by astridx** [comment]
> Is this setup in draft mode?

Yes, because I am not sure on point 1 and 3 in the in the first command here.

**Comment by astridx** [comment]
This is now ready for review.

Mainly:
I am now sure, that it makes no difference that in the tape tests the variable t is given as parameter for our Jest tests.
I put `window.location.hash` handling in `beforeEach `and `afterEach`.

**Review by HarelM — APPROVED** [review]

**Pull request merged**